### PR TITLE
feat(route): add アニメ新番組

### LINKED ISSUE #10291: アニメ新番組
### 类型

二次元

### 网站地址

https://bangumi.online/

### 网站描述

動畫新番

### 需要生成什么内容？

title 標題
link 連結
media:content 圖片預覽

### 额外描述

_No response_

### 这不是重复的 RSS 请求

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues) 和 [pull requests](https://github.com/DIYgod/RSSHub/pulls)，以确保该 RSS 尚未被请求。

### PR DESCRIPTION
**Pull request by devinryu**
## 该 PR 相关 Issue / Involved issue

Close #10291 

## 完整路由地址 / Example for the proposed route(s)

```routes
/bangumi/online
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/bangumionline/index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+module.exports = async (ctx) => {
+    const url = 'https://bangumi.online/api/home';
+
+    const response = await got({
+        method: 'post',
+        url,
+        headers: {},
+    });
+
+    const list = JSON.parse(response.body).data.list;
```
Comment (TonyRL) [inline lib/v2/bangumionline/index.js:12]:
```suggestion
    const response = await got(url);

    const list = response.data.data.list;
```

File: lib/v2/bangumionline/index.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    const items = list.map((item) => ({
+        title: `${item.title_zh} - 第 ${item.volume} 集`,
+        description: `<img src='https:${item.cover}'>`,
```
Comment (TonyRL) [inline lib/v2/bangumionline/index.js:16]: https://docs.rsshub.app/joinus/script-standard.html#template

File: lib/v2/bangumionline/index.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        link: `https://bangumi.online/watch/${item.vid}`,
+    }));
+
+    ctx.state.data = {
+        title: `アニメ新番組`,
+        link: `https://bangumi.online/`,
```
Comment (TonyRL) [inline lib/v2/bangumionline/index.js:22]:
```suggestion
        title: 'アニメ新番組',
        link: 'https://bangumi.online',
```

File: lib/v2/bangumionline/radar.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'bangumi.online': {
+        _name: 'アニメ新番組',
+        '.': [
+            {
+                title: 'bangumi online',
```
Comment (TonyRL) [inline lib/v2/bangumionline/radar.js:6]:
```suggestion
                title: '當季新番',
```
This should match the H3 heading of the docs

File: lib/v2/bangumionline/radar.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+                docs: 'https://docs.rsshub.app/anime.html#shibangumi-online',
```
Comment (TonyRL) [inline lib/v2/bangumionline/radar.js:7]: Wrong url hash

File: docs/anime.md [hunk 6]
```diff
@@ -652,3 +652,9 @@ Sources
 ### 最新汉化
 
 <Route author="junfengP" example="/zdfx" path="/zdfx"/>
+
+## アニメ新番組
+
+### 當季新番
+
+<Route author="devinmugen" example="/bangumionline/updates" path="/bangumionline/updates"/>
```
Comment (TonyRL) [inline docs/anime.md:660]: I would suggest using the namespace `bangumi` as the domain name is `bangumi`.online but not `bangumionline`.com/.org, etc.

File: lib/v2/bangumionline/index.js (hunk 2 above)
Comment (devinmugen) [reply lib/v2/bangumionline/index.js]: Thank you, art templates are already in use

File: lib/v2/bangumionline/index.js (hunk 1 above)
Comment (devinmugen) [reply lib/v2/bangumionline/index.js]: Using `const response = await got(url);` will return response code 404.

Comment (TonyRL) [reply lib/v2/bangumionline/index.js]: `const response = await got.post(url);`

Comment (devinmugen) [reply lib/v2/bangumionline/index.js]: Thank you ~

File: lib/v2/bangumionline/index.js (hunk 3 above)
Comment (devinmugen) [reply lib/v2/bangumionline/index.js]: Changed

File: lib/v2/bangumionline/radar.js (hunk 4 above)
Comment (devinmugen) [reply lib/v2/bangumionline/radar.js]: Changed

File: lib/v2/bangumionline/radar.js (hunk 5 above)
Comment (devinmugen) [reply lib/v2/bangumionline/radar.js]: I'm not sure if I've amended it correctly this time.
`https://docs.rsshub.app/anime.html#アニメ-xin-fan-zu-dang-ji-xin-fan`

File: docs/anime.md (hunk 6 above)
Comment (devinmugen) [reply docs/anime.md:660]: `bangumi` usually means `bangumi.tv`, so I have added `online` to distinguish it.

Comment (TonyRL) [reply docs/anime.md:660]: You can use something like `/bangumi/online/updates`. In fact, `/bangumi` is used for `bangumi.moe` currently, and routes related to `bangumi.tv` haves their own category like `/bangumi/calendar/today` or `/bangumi/subject`

Comment (mugen0x) [reply docs/anime.md:660]: Under V2 does `bangumionline` need to be merged under `bangumi` in order to use `bangumi/online`? Or is there another way?

Comment (devinmugen) [reply docs/anime.md:660]: Under V2 does `bangumionline` need to be merged under `bangumi` in order to use `bangumi/online`? Or is there another way?

Comment (TonyRL) [reply docs/anime.md:660]: Yes, it needs to be merged in order to use `/bangumi/online`

Comment (devinmugen) [reply docs/anime.md]: I have changed the route to `bangumi/online`. 
The github-actions bot shows a failure, but it has been successfully deployed on Vercel.
https://rssh-ub-devinmugen.vercel.app/bangumi/online

**Pull request merged**